api-server/src/server/boot/challenge.js

```diff
@@ -373,6 +422,52 @@ function backendChallengeCompleted(req, res, next) {
     .subscribe(() => {}, next);
 }
 
+function saveChallenge(req, res, next) {
```

Comment:
Can you DRY this out?  There's a fair bit of duplication between `buildUserUpdate` and this.

Reply:
resolved locally - will be added on next push